src/FloatingButton/FloatingButton.tsx

```diff
@@ -0,0 +1,199 @@
+import * as classnames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+
+import { Button } from "../Button/Button";
+import FlexBox from "../flex/FlexBox";
+
+import "./FloatingButton.less";
+
+const cssClass = {
+  ACTIVE: "FloatingButton--active",
+  inactive: positionY => `FloatingButton--inactive--${positionY}`,
+
+  propStyle: prop => `FloatingButton--${prop}`,
+
+  BUTTON: "FloatingButton--button",
+  CONTAINER: "FloatingButton--container",
+  REMOVE: "FloatingButton--remove",
+  SUB_ELEMENT: "FloatingButton--subElement",
+};
+
+const propTypes = {
+  animate: PropTypes.bool,
+
+  additionalButtons: PropTypes.array,
+
+  bgColor: PropTypes.string,
+  className: PropTypes.string,
+  closeLabel: PropTypes.node,
+  label: PropTypes.node,
+
+  onClick: PropTypes.func,
+
+  offsetX: PropTypes.number,
+  offsetY: PropTypes.number,
+  positionX: PropTypes.oneOf(["left", "center", "right"]),
+  positionY: PropTypes.oneOf(["bottom", "top"]),
+  size: PropTypes.string,
+};
+
+const defaultProps = {
+  animate: true,
+  label: "+",
+  offsetX: 30,
+  offsetY: 30,
+  positionX: "right",
+  positionY: "bottom",
+};
+
+export default class FloatingButton extends React.PureComponent {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+  static cssClass = cssClass;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      active: false,
+      display: false,
+    };
+  }
+
+  horizontalStyle() {
+    const { offsetX, positionX } = this.props;
+    if (positionX === "center") return {};
+    return positionX === "left"
+      ? {
+          left: `${offsetX}px`,
+        }
+      : {
+          // default to right if a bad positionX input provided
+          right: `${offsetX}px`,
+        };
+  }
+
+  verticalStyle() {
+    const { offsetY, positionY } = this.props;
+    return positionY === "top"
+      ? {
+          top: `${offsetY}px`,
+        }
+      : {
+          // default to bottom orientation
+          bottom: `${offsetY}px`,
+        };
+  }
+
+  mainButtonHandler = () => {
+    const { additionalButtons, animate, onClick } = this.props;
+    const { active } = this.state;
+    if (additionalButtons && additionalButtons.length) {
+      // main button activates additional buttons
+      if (active) {
+        // stop displaying when transition to inactive state complete
+        this.setState({ active: false, display: true });
```

Comment:
Why do we need both an `active` state and a `display` state? Aren't these tracking the same thing? Can we use CSS transitions to handle the animations so we can avoid the race conditions you mention below?

Reply:
this is a little hacky. It's my way of persisting display while the transition runs. I don't believe it's possible to do this in pure css because the transition property doesn't act on display

Reply:
What effect are you trying to accomplish? It looks like you're already animating the buttons away with CSS transitions without using this hack? 🤔 Can you use `pointer-events: none;`?

Reply:
I had to make the container pointer-events: none and subElements auto (if being displayed), but this worked